TBA-245 Update transfer sdk documentation

OlehSyzonov opened the pull request:
* Renamed Transfer Client -> Transfer SDK
 - Removed dependencies to Web3 library
 * Updated  solanaNEONTransferTransaction parameters
 - Removed "Alternative libraries" 
 - Hidden "React Demo App" due to bug FRONT-429 
 * Updated "2.1 Transfer NEON"
 * Restructured "2.3 Multi-token gas fee" -> "2.2 Transfer SOL"
 * Updated code examples in "2.2 Transfer SOL"
 * Updated code examples in "2.3 Transfer ERC-20 tokens" + Added reference to https://sdk.neonevm.org/
 * Fixed dead link: https://docs.solana.com/storage_rent_economics -> https://solana.com/docs/core/fees#rent

OlehSyzonov committed:
Update neon_transfer.mdx

* Renamed Transfer Client -> Transfer SDK
- Removed dependencies to Web3 library
* Updated  solanaNEONTransferTransaction parameters

OlehSyzonov committed:
Update neon_transfer.mdx

- Removed "Alternative libraries"
- Hidden "React Demo App" due to bug FRONT-429
* Updated "2.1 Transfer NEON"

OlehSyzonov committed:
Update neon_transfer.mdx

* Restructured "2.3 Multi-token gas fee" -> "2.2 Transfer SOL"
* Updated code examples in "2.2 Transfer SOL"
* Updated code examples in "2.3 Transfer ERC-20 tokens" 
+ Added reference to https://sdk.neonevm.org/

OlehSyzonov committed:
Update gas_fees.mdx

Fixed dead link: https://docs.solana.com/storage_rent_economics ->  https://solana.com/docs/core/fees#rent

eubash commented on docs/developing/integrate/neon_transfer.mdx:
```diff
@@ -112,16 +113,16 @@ import {
 };
 
-const transaction = await solanaNEONTransferTransaction(solanaWallet.publicKey, neonWallet.address, neonEvmProgram, neonTokenMint, neonToken, amount, chainId);
+const transaction = await solanaNEONTransferTransaction(solanaWallet, neonWallet, neonEvmProgram, neonTokenMint, neonToken, amount, chainId);
```
Change to:
```
const transaction = await solanaNEONTransferTransaction({
  solanaWallet,
  neonWallet,
  neonEvmProgram,
  neonTokenMint,
  token: neonToken,
  amount,
  chainId: neonChainId
});
```

eubash commented on docs/developing/integrate/neon_transfer.mdx:
```diff
@@ -112,16 +113,16 @@ import {
 
-const transaction = await solanaNEONTransferTransaction(solanaWallet.publicKey, neonWallet.address, neonEvmProgram, neonTokenMint, neonToken, amount, chainId);
+const transaction = await solanaNEONTransferTransaction(solanaWallet, neonWallet, neonEvmProgram, neonTokenMint, neonToken, amount, chainId);
 const signature = await sendSolanaTransaction(connection, transaction, [toSigner(solanaWallet)]);
```
Change this `const signature = await sendSolanaTransaction(connection, transaction, [toSigner(solanaWallet)]);` to:
```
const signers = [toSigner(solanaWallet)];
transaction.recentBlockhash = (await connection.getLatestBlockhash('finalized')).blockhash; // Network blockhash
transaction.sign(...signers);
const signature = await connection.sendRawTransaction(transaction.serialize());
```

eubash commented on docs/developing/integrate/neon_transfer.mdx:
```diff
@@ -138,96 +139,33 @@ import {
-  const solanaTransaction = createAssociatedTokenAccountTransaction(solanaWallet.publicKey, mint, associatedToken);
-  const signature = await sendSolanaTransaction(connection, solanaTransaction, [toSigner(solanaWallet)]);
-  console.log(`Associated token account`, signature);
+const transaction = await neonNeonTransactionEthers({ provider, neonWallet.address, NEON_TRANSFER_CONTRACT_DEVNET, solanaWallet, amount });
```
Change this to:
```
const transaction = await neonNeonTransactionEthers({
  provider,
  from: neonWallet.address,
  to: NEON_TRANSFER_CONTRACT_DEVNET,
  solanaWallet,
  amount
});
```

eubash reviewed (changes requested):
@OlehSyzonov, please check all changes.

OlehSyzonov commented on docs/developing/integrate/neon_transfer.mdx:
```diff
@@ -275,42 +223,116 @@ console.log(signature);
-import { neonNeonTransactionWeb3 } from '@neonevm/token-transfer-web3';
-import { SOL_TRANSFER_CONTRACT_DEVNET, SPLToken } from '@neonevm/token-transfer-core';
+import {
+  SOL_TRANSFER_CONTRACT_DEVNET,
```
This piece of code is about sending SOL tokens from Neon EVM to Solana

eubash replied:
If you send SOL tokens, your SPL token should have: `neonChainId`, `solTokenMint.toBase58()` and you don't need NEON_TOKEN_MINT_DECIMALS.
Transaction in that case would be:
`const transaction = await neonNeonTransactionEthers({ provider, from: neonWallet.address, to: SOL_TRANSFER_CONTRACT_DEVNET, solanaWallet: solanaWallet.publicKey, amount });`

OlehSyzonov replied:
Would be such token definition correct:
`const neonToken: SPLToken = {
  chainId: neonChainId,
  address_spl: solTokenMint.toBase58(),
  address: '0xc7Fc9b46e479c5Cb42f6C458D1881e55E6B7986c',
  decimals: 9,
  name: 'Solana SOL',
  symbol: 'SOL',
  logoURI: 'https://raw.githubusercontent.com/neonlabsorg/token-list/master/assets/solana-sol-logo.svg'
};`

eubash replied:
It would be:
```
const neonToken: SPLToken = {
chainId: solChainId,
address_spl: solTokenMint.toBase58(),
address: '0xc7Fc9b46e479c5Cb42f6C458D1881e55E6B7986c',
decimals: 9,
name: 'Solana SOL',
symbol: 'SOL',
logoURI: 'https://raw.githubusercontent.com/neonlabsorg/token-list/master/assets/solana-sol-logo.svg'
};
```

dariaag commented:
this page should not be inside "Integrate" folder as it is not an integration, it breaks the urls

OlehSyzonov committed:
Update neon_transfer.mdx

* Update reviewed code examples

OlehSyzonov committed:
Rename docs/developing/integrate/neon_transfer.mdx to docs/token_transferring/neon_transfer.mdx

> Moved to token_transfering folder

OlehSyzonov committed:
Update sidebars.js

+ Added category '🛰 Transfer Tokens'

OlehSyzonov committed:
Update sidebars.js

* fixed broken link

OlehSyzonov commented:
> this page should not be inside "Integrate" folder as it is not an integration, it breaks the urls

Moved to "token_transferring"
Please se if you can approve the PR.

OlehSyzonov committed:
Update token_list.md

* Fixed broken link

eubash commented on docs/token_transferring/neon_transfer.mdx:
```diff
@@ -0,0 +1,371 @@
+import { JsonRpcProvider } from 'ethers';
+
+const neonNeonEvmUrl = `https://devnet.neonevm.org`;
+const solNeonEvmUrl = `https://devnet.neonevm.org/solana/sol`;
```
The right url here is: `https://devnet.neonevm.org/sol`

OlehSyzonov committed:
Update interacting_with_spl_tokens.md

* Updated link to Transfer SDR doc

OlehSyzonov committed:
Update neon-brief-faq.md

* Updated link to Transfer SDR doc

eubash commented on docs/token_transferring/neon_transfer.mdx:
```diff
@@ -0,0 +1,371 @@
+  logoURI: 'https://raw.githubusercontent.com/neonlabsorg/token-list/master/assets/solana-sol-logo.svg'
+};
+
+const transaction = await neonNeonTransactionEthers({ provider, from: neonWallet.address, SOL_TRANSFER_CONTRACT_DEVNET, solanaWallet: solanaWallet.publicKey, amount });
```
here **to** was missed: 
`const transaction = await neonNeonTransactionEthers({ provider, from: neonWallet.address, to: SOL_TRANSFER_CONTRACT_DEVNET, solanaWallet: solanaWallet.publicKey, amount });`

OlehSyzonov committed:
Update neonpass.md

* Updated link to Transfer SDR doc

OlehSyzonov committed:
Update neonpass_usage.md

* Updated link to Transfer SDR doc

eubash commented on docs/token_transferring/neon_transfer.mdx:
```diff
@@ -0,0 +1,371 @@
+transaction.recentBlockhash = (await connection.getLatestBlockhash('finalized')).blockhash; // Network blockhash
+transaction.sign(...signers);
+const signature = await connection.sendRawTransaction(transaction.serialize());
+//Create and send NEON transaction
```
We can also add here: 
`const provider = new JsonRpcProvider(proxyUrl);`

eubash reviewed (changes requested):
Check comments please

OlehSyzonov committed:
Update neon_transfer.mdx

* Updates to the code examples

OlehSyzonov commented:
> Check comments please

Done

OlehSyzonov committed:
Update sidebars.js

* Restructured menu

OlehSyzonov committed:
Update neonpass_usage.md

* Changed page title

eubash commented on docs/token_transferring/neon_transfer.mdx:
````diff
@@ -0,0 +1,373 @@
+```javascript
+import { solanaSOLTransferTransaction, SPLToken } from '@neonevm/token-transfer-core';
+
+const neonToken: SPLToken = {
````
Here we should change `neonToken: SPLToken =` to `solToken: SPLToken =`

eubash commented on docs/token_transferring/neon_transfer.mdx:
```diff
@@ -0,0 +1,373 @@
+  SOL_TRANSFER_CONTRACT_DEVNET,
+} from '@neonevm/token-transfer-core';
+
+const neonToken: SPLToken = {
```
Here we don't need to have solToken, as. it is native, we don't pass it in the transaction. You can remove it. Remove all this:
```
const neonToken: SPLToken = {
  chainId: solChainId,
  address_spl: solTokenMint.toBase58(),
  address: '0xc7Fc9b46e479c5Cb42f6C458D1881e55E6B7986c',
  decimals: 9,
  name: 'Solana SOL',
  symbol: 'SOL',
  logoURI: 'https://raw.githubusercontent.com/neonlabsorg/token-list/master/assets/solana-sol-logo.svg'
};
```

eubash commented on docs/token_transferring/neon_transfer.mdx:
````diff
@@ -0,0 +1,373 @@
+```javascript
+import tokenList from 'token-list/tokenlist.json';
+
+const tokens = tokenList.tokens.filter((token) => token.chainId === CHAIN_ID);
````
Add `const token = tokens[0];`

eubash reviewed (changes requested):
@OlehSyzonov, check comments please.

OlehSyzonov commented:
> @OlehSyzonov, check comments please.

Done
Thank you!

OlehSyzonov committed:
Update neon_transfer.mdx

* Code improvements

OlehSyzonov committed:
Update neon_transfer.mdx

The SDK usage guideline is substituted with the reference to the Transfer SDK Documentation https://sdk.neonevm.org/index.html

OlehSyzonov committed:
Update neon_transfer.mdx

* Fix of typo
